### PR TITLE
DEV-5767 Added the ability to use async to upsert transactions for mu…

### PR DESCRIPTION
**Pull request by MikeMcGarry**
…ltiple portfolios simultaneously while also batching each request to allow for more than 10,000 transactions to be upserted in a single call

### CODE REVIEW

File: tests/integration/cocoon/test_cocoon_transactions.py [hunk 1]
```diff
@@ -166,6 +199,7 @@ def test_load_from_data_frame_transactions_success(
             identifier_mapping=identifier_mapping,
             property_columns=property_columns,
             properties_scope=properties_scope,
+            batch_size=batch_size
         )
 
         self.assertTrue(
```
Comment (rizwansaeed) [inline tests/integration/cocoon/test_cocoon_transactions.py]: Just looking at the assert, does this pick up failures correctly i.e. when the request fails do we get back an empty (valid) version which makes this pass or do we not get that back which means what we have is currently ok?

File: lusidtools/cocoon/cocoon.py [hunk 2]
```diff
@@ -418,10 +418,10 @@ def load_portfolio_batch(
 
             sync_batches = [
```
Comment (rizwansaeed) [inline lusidtools/cocoon/cocoon.py]: Add a comment to explain what this is doing

File: tests/integration/cocoon/test_cocoon_transactions.py (hunk 1 above)
Comment (MikeMcGarry) [reply tests/integration/cocoon/test_cocoon_transactions.py]: On a failure we get an exception back.

Comment (MikeMcGarry) [reply tests/integration/cocoon/test_cocoon_transactions.py]: However this only checks successes and you are correct that we need to check that the number of successes is greater than 0.

File: lusidtools/cocoon/cocoon.py (hunk 2 above)
Comment (MikeMcGarry) [reply lusidtools/cocoon/cocoon.py]: Added

**Pull request merged**